test/unit/tests/002recordExpressionParser.test.js

```diff
@@ -99,6 +108,17 @@ describe('RecordExpressionParser Test', () => {
     { q: 'isEmpty(tree_height)', r: false },
     // remarks is empty
     { q: 'isEmpty(remarks)', r: true },
+    // plot count is 3
+    { q: 'plot.length', r: 3 },
+    // access multiple entities with index
+    { q: 'plot[0].plot_id', r: 1 },
+    { q: 'plot[1].plot_id', r: 2 },
+    { q: 'plot[2].plot_id', r: 3 },
+    { q: 'plot[4].plot_id', r: null },
+    // plot_multiple_number counts
+    { q: 'plot[0].plot_multiple_number.length', r: 2 },
+    { q: 'plot[1].plot_multiple_number.length', r: null },
+    { q: 'plot[2].plot_multiple_number.length', r: 1 },
```

Comment:
we should also test that 
- 'string'.length exist
- number.length throws an error
- 'string'[idx] returns the character at the index idx

Reply:
it will be done in another PR